### ISSUE TITLE
Synchronizing Tasks with google fails

### ISSUE THREAD

**Issue by Znrl** [aluxnimm/outlookcaldavsynchronizer#280]
It's a fresh install of Outlook CalDav Synchronizer 3.8.2
With Outlook 365, I updadet it just before installing
On Windows 10
Syncing with google

Synchronizing the calendar seems to work.

But for Tasks I get:

```<?xml version="1.0" encoding="utf-16"?>
<SynchronizationReport xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xmlns:xsd="http://www.w3.org/2001/XMLSchema">
  <ProfileName>Aufgaben - some-name</ProfileName>
  <ProfileId>52b9c95c-b854-491c-894a-d3df9fb69e47</ProfileId>
  <StartTime>2020-07-02T18:48:08.778467Z</StartTime>
  <LoadErrors />
  <EntitySynchronizationReports />
  <ExceptionThatLeadToAbortion>System.NullReferenceException: Der Objektverweis wurde nicht auf eine Objektinstanz festgelegt.
   bei CalDavSynchronizer.Implementation.GoogleTasks.GoogleTaskRepository.&lt;GetAllVersions&gt;d__7.MoveNext()
--- Ende der Stapelüberwachung vom vorhergehenden Ort, an dem die Ausnahme ausgelöst wurde ---
   bei System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)
   bei System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   bei GenSync.EntityRepositories.VersionAwareToStateAwareEntityRepositoryAdapter`4.&lt;GetFullRepositoryState&gt;d__5.MoveNext()
--- Ende der Stapelüberwachung vom vorhergehenden Ort, an dem die Ausnahme ausgelöst wurde ---
   bei System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)
   bei System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   bei System.Runtime.CompilerServices.TaskAwaiter.ValidateEnd(Task task)
   bei GenSync.Synchronization.Synchronizer`11.&lt;Synchronize&gt;d__27.MoveNext()
--- Ende der Stapelüberwachung vom vorhergehenden Ort, an dem die Ausnahme ausgelöst wurde ---
   bei System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)
   bei System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   bei GenSync.Synchronization.NullContextSynchronizerDecorator`6.&lt;Synchronize&gt;d__2.MoveNext()
--- Ende der Stapelüberwachung vom vorhergehenden Ort, an dem die Ausnahme ausgelöst wurde ---
   bei System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)
   bei System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   bei CalDavSynchronizer.Scheduling.SynchronizationProfileRunner.&lt;RunAndRescheduleNoThrow&gt;d__22.MoveNext()</ExceptionThatLeadToAbortion>
  <ConsiderExceptionThatLeadToAbortionAsWarning>false</ConsiderExceptionThatLeadToAbortionAsWarning>
  <Duration>00:00:00.9574403</Duration>
</SynchronizationReport>
```

Any Idea what I could try?

**Comment by Znrl** [comment]
To add a little bit...
I think this happens when there are no tasks in Google.
I did not receive an error now when I added a test task.
There where no tasks in Outlook and no in Google when I tried this.

So maybe there needs to be some other way to handle "no tasks" than to have an error.

**Comment by aluxnimm** [comment]
Yes we identified the bug and will update google task api library in the next release.

Duplicate of
https://sourceforge.net/p/outlookcaldavsynchronizer/tickets/1377/

**Comment by aluxnimm** [comment]
fixed in 4.0.0
